### PR TITLE
TEST-#6505: update python version for ASV benchmarks on HDK

### LINKED ISSUE #6505: Update python version for ASV benchmarks on HDK
python 3.7 is outdated version

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6505 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: asv_bench/asv.conf.json [hunk 1]
```diff
@@ -53,6 +53,10 @@
     // the base URL to show a commit for the project.
     "show_commit_url": "https://github.com/modin-project/modin/commit/",
 
+    // The Pythons you'd like to test against.  If not provided, defaults
+    // to the current version of Python used to run `asv`.
+    "pythons": ["3.8"],
```
Comment (anmyachev) [inline asv_bench/asv.conf.json:58]: Python 3.8 was used before, but it's better to set the version explicitly, as in other configurations.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -265,6 +265,10 @@ jobs:
               sed -i 's/def time_groupby_agg_nunique(self, \*args, \*\*kwargs):/# def time_groupby_agg_nunique(self, *args, **kwargs):/g' benchmarks/hdk/benchmarks.py
               sed -i 's/execute(self.df.groupby(by=self.groupby_columns).agg("nunique"))/# execute(self.df.groupby(by=self.groupby_columns).agg("nunique"))/g' benchmarks/hdk/benchmarks.py
 
+              # Otherwise, ASV considers that the environment has already been created, although ASV command is run for another config,
+              # which requires the creation of a completely new environment. This step will be required after removing the manual environment setup step.
+              rm -f -R .asv/env/
```
Comment (anmyachev) [inline .github/workflows/ci.yml:270]: This change will make future changes easier.

File: asv_bench/asv.conf.json (hunk 1 above)
Comment (YarShev) [reply asv_bench/asv.conf.json:58]: Do we need the code duplication below?

Comment (anmyachev) [reply asv_bench/asv.conf.json:58]: Removed

**Review by YarShev — APPROVED** [review]

**Comment by anmyachev** [comment]
@YarShev let's merge

**Pull request merged**